### PR TITLE
Implemented the Gnome Sort Completely.

### LINKED ISSUE #1831: Implementation of Gnome Sort
I want to work on it in C, C++, Java, and Python. Please assign this issue to me @singh-shreya6 @raghav-dalmia @asha15

### PR DESCRIPTION
**Pull request by srimani-programmer**
Fixes #1831

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Implemented the Gnome_Sort Algorithm From Scratch.

#### Languages Used:
- C
- C++
- Java
- Python

#### Files Added:
-  Gnome_Sort.c
-  Gnome_Sort.cpp
- Gnome_Sort.java
- Gnome_Sort.py

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

File: Gnome_Sort/Gnome_Sort.c [hunk 1]
```diff
@@ -0,0 +1,64 @@
+#include <stdio.h> 
+ 
+  
+// A function which is used to sort the algorithm using gnome sort 
+void gnome_sort(int arr[], int n) 
+{   
+    // 
+        int index = 0;
```
Comment (jainaman224) [inline Gnome_Sort/Gnome_Sort.c]: Fix indentation. Please have a look at checklist.

File: Gnome_Sort/Gnome_Sort.c [hunk 2]
```diff
@@ -0,0 +1,64 @@
+#include <stdio.h> 
+ 
+  
+// A function which is used to sort the algorithm using gnome sort 
+void gnome_sort(int arr[], int n) 
+{   
+    // 
+        int index = 0; 
+  
+        while (index < n){
```
Comment (jainaman224) [inline Gnome_Sort/Gnome_Sort.c]: Add space before `{`.

File: Gnome_Sort/Gnome_Sort.c [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+   
+} 
+  
+  
+// Sample Test cases to test above function. 
+int main() 
+{ 
+    int n;
+    printf("Enter the size of an Array: ");
+    scanf("%d",&n);
```
Comment (jainaman224) [inline Gnome_Sort/Gnome_Sort.c]: Add space after `,`, `;`.

File: Gnome_Sort/Gnome_Sort.c [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    7
+
+    OUTPUT:
+    -5
+    1
+    2
+    3
+    7
+
+*/
```
Comment (jainaman224) [inline Gnome_Sort/Gnome_Sort.c]: Add a single new line at the end of file.

File: Gnome_Sort/Gnome_Sort.cpp [hunk 5]
```diff
@@ -0,0 +1,64 @@
+#include <iostream> 
+
+using namespace std; 
+  
+// A function which is used to sort the algorithm using gnome sort 
+void gnome_sort(int arr[], int n) 
+{   
+    //
```
Comment (jainaman224) [inline Gnome_Sort/Gnome_Sort.cpp]: Empty comment?

File: Gnome_Sort/Gnome_Sort.cpp [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+} 
+  
+  
+// Sample Test cases to test above function. 
+int main() 
+{ 
+    
+    int n;
+    cout<<"Enter the size of an Array: ";
+    cin>>n;
```
Comment (jainaman224) [inline Gnome_Sort/Gnome_Sort.cpp]: Add space around `>>`, `<<`, `=`, `<`, `==`, `+=`.

File: Gnome_Sort/Gnome_Sort.java [hunk 7]
```diff
@@ -0,0 +1,60 @@
+import java.util.*;
+class GnomeSort {
```
Comment (jainaman224) [inline Gnome_Sort/Gnome_Sort.java]: Add a new line above this.

**Comment by srimani-programmer** [comment]
Please merge this branch, sir.

**Pull request merged**